src/core/Rule.ml

```diff
@@ -28,7 +28,46 @@ open Ppx_hash_lib.Std.Hash.Builtin
  * See also Mini_rule.ml where formula and many other features disappear.
  *)
 
-type rule_id = string [@@deriving show]
+(*
+   A rule ID is a string. This creates a dedicated type to clarify interfaces
+   and error messages. The coercion operator can be used as an alternative
+   to 'to_string x': '(x :> string)' to obtain a plain string.
+*)
+module ID : sig
+  type t = private string [@@deriving show, eq]
+
+  val to_string : t -> string
+  val of_string : string -> t
+  val to_string_list : t list -> string list
+  val of_string_list : string list -> t list
+  val compare : t -> t -> int
+
+  (*
+     Rule ids are prepended with the `path.to.the.rules.file.`, so
+     when comparing a rule (r) with the rule to be included or excluded,
+     allow for a preceding path
+
+       "path.to.foo.bar" ~suffix:"foo.bar" -> true
+       "foo.bar" ~suffix:"foo.bar" -> true
+       "xfoo.bar" ~suffix:"foo.bar" -> false
+
+     TODO we may want to allow globs for rules
```

Comment:
we do already, see the recent extension @emjin did for extract mode where people want to specify the set of rules one can with the extract rule.

Reply:
This is an old comment I copied from somewhere else. Removing.